api/v1/openapi.json

```diff
@@ -0,0 +1,1115 @@
+                                                "schema": {
+                                                        "$ref": "#/definitions/Error"
+                                                }
+                                        }
+				}
+			}
+		},
+                "/service/{id}": {
+			"get": {
+				"summary": "Retrieve configuration of a service",
+				"tags": [ "service" ],
+				"parameters": [
+					{ "$ref": "#/parameters/service-id" },
+					{ "$ref": "#/parameters/service-address" }
+				],
+				"responses": {
+					"200": {
+                                                "description": "Success",
+                                                "schema": {
+							"$ref": "#/definitions/Service"
+                                                }
+					},
+                                        "400": {
+                                                "description": "Address or ID parameter required"
+                                        },
+                                        "404": {
+                                                "description": "Service not found"
+                                        }
+				}
+			},
+			"put": {
+				"summary": "Create or update service",
+				"tags": [ "service" ],
+				"parameters": [
+					{ "$ref": "#/parameters/service-id" },
+					{ "$ref": "#/parameters/service-config" }
+				],
+				"responses": {
+					"201": {
+                                                "description": "Success"
+					},
+                                        "default": {
+                                                "description": "Unexpected error",
+                                                "schema": {
+                                                        "$ref": "#/definitions/Error"
+                                                }
+                                        }
+				}
+			},
+			"delete": {
+				"summary": "Delete a service",
+				"tags": [ "service" ],
+				"parameters": [
+					{ "$ref": "#/parameters/service-id" },
+					{ "$ref": "#/parameters/service-address" }
+				],
+				"responses": {
+					"200": {
+                                                "description": "Success"
+					},
+                                        "404": {
+                                                "description": "Service not found",
+                                                "schema": {
+                                                        "$ref": "#/definitions/Error"
+                                                }
+                                        }
+				}
+			}
+		}
+        },
+	"parameters": {
+		"endpoint-id": {
+			"name": "id",
+			"description": "Endpoint ID",
+			"in": "query",
+			"required": true,
+			"type": "string"
+		},
+		"endpoint-id-type": {
+			"name": "type",
+			"description": "Type of endpoint ID",
+			"in": "query",
+			"type": "string",
+			"default": "cilium",
+			"enum": [
+				"cilium",
+				"docker-id",
+				"docker-endpoint"
```

Comment:
See comment on "id" from below. Could we make that more consistent or better structured? F.e. "docker-id" vs. "docker-endpoint" as "endpoint-id-type" seems perhaps weird. "cilium" is mentioned here but implicitly id in the EP. If we could formalize that in a more structured way, would be great.